### PR TITLE
fix flow animation for svg

### PR DESCRIPTION
**Pull request by cool-runnings**

### CODE REVIEW

**Comment by joltcoke** [comment]
Label: Bugfix
Milestone: 0.9.0

**Comment by ChristianMayer** [comment]
I'm not too happy about the chosen solution as it is reactivating a mechanism that was removed.
Currently I'm still reviewing it.

File: src/lib/templateengine.js [hunk 1]
```diff
@@ -402,10 +404,10 @@ function TemplateEngine( undefined ) {
...
   
   this.addAddress = function( address, id ) {
-    if( address in ga_list )
-      ga_list[ address ].push( id );
-    else
+    if ( !(address in ga_list) )
       ga_list[ address ] = [ id ];
+    else if ( $.inArray(id,ga_list[address]) == -1 )
+      ga_list[ address ].push( id );
   };
```
Comment (ChristianMayer) [inline src/lib/templateengine.js]: What is the reason for this change? Replacing a native JS function by an indirect jQuery function seems like adding only overhead

File: src/lib/trick-o-matic.js [hunk 2]
```diff
@@ -139,4 +141,26 @@ define([ 'jquery' ], function( $ ) {
...
+    $(pipes).each(function(idx, pipe) {
+      var activeValues = $(pipe).data('cometvisu-active');
+      if ( !activeValues ) { return true; }
+      $(activeValues.split(' ')).each(function(idx, ga) {
+        templateEngine.addAddress(ga, 'svg');
+      });
+    });    
+  });
+  jQuery.ajaxSetup({async:true});
+};
```
Comment (ChristianMayer) [inline src/lib/trick-o-matic.js]: Do I interpret that correctly:
You are reloading the SVG backdrop just to parse it for the GAs? And are using a synchronous call for that?

If that's true I'm sorry to tell that this are two no gos for me:
- the data is already loaded and parsed in the browser and should be accessed through the DOM
- synchronous is considered bad practise in JavaScript and only very special occasions allow it

**Comment by ChristianMayer** [comment]
Please see https://github.com/CometVisu/CometVisu/pull/226 for a different bug fix

**Comment by cool-runnings** [comment]
Maybe this solution wasn't the best idea. I'll have a look for your approach.
